platform-hub-api/spec/controllers/me_controller_spec.rb

```diff
@@ -42,6 +42,51 @@
     end
   end
 
+  describe '#delete_identity' do
+    it_behaves_like 'authenticated' do
+      before do
+        create(:identity,
+          user: current_user,
+          external_id: '1234567',
+          external_username: test_auth_payload.preferred_username,
+          external_name: test_auth_payload.name,
+          external_email: test_auth_payload.email
+        )
+      end
+
+      it 'should delete an identity successfully if it exists for the current user' do
+        expect(Audit.count).to eq 0
+        delete :delete_identity, params: {service: 'github'}
+        expect(response).to be_success
+        expect(current_user.identity('github')).to be nil
+        expect(Audit.count).to eq 1
+        expect(Audit.first.action).to eq 'delete_identity'
+        expect(Audit.first.user.id).to eq current_user_id
+      end
+
+      it 'should gracefully fail to delete an identity that does not exist' do
+        delete :delete_identity, params: {service: 'kubernetes'}
+        expect(response).to have_http_status(404)
+        expect(json_response['error']['message']).to eq "Resource not found"
+        expect(Audit.count).to eq 0
+      end
+
+      it 'should gracefully fail to delete an identity if called twice' do
+        expect(Audit.count).to eq 0
+        delete :delete_identity, params: {service: 'github'}
+        expect(response).to be_success
+        expect(current_user.identity('github')).to be nil
+        expect(Audit.count).to eq 1
+        expect(Audit.first.action).to eq 'delete_identity'
+        expect(Audit.first.user.id).to eq current_user_id
+        delete :delete_identity, params: {service: 'github'}
```

Comment:
Likewise here, and the two other usages above – should use `@identity.provider` to couple the value to the actual factory object.